### PR TITLE
feat: Add put/patch state support

### LINKED ISSUE #164: Add put/patch state api methods in the HoverflyClient and in Hoverfly
As said in #147, this is the first step to be able to execute requests multiple times with different responses.

### PR DESCRIPTION
**Pull request by ia3andy**
Add:
- HoverflyClient.setState(StateView)
- HoverflyCient.udpateState(StateView)
- HoverflyCient.getState():StateView

- Hoverfly.setState(Map<String, String>)
- Hoverfly.udpateState(Map<String, String>)
- Hoverfly.getState():Map<String, String>

- HoverflyRule.setState(Map<String, String>)
- HoverflyRule.udpateState(Map<String, String>)
- HoverflyRule.getState():Map<String, String>

Changes:

- HoverflyClient.deleteStates -> Hoverfly.deleteState
- Hoverfly.resetStates -> Hoverfly.resetState
- HoverflyRule.resetStates -> HoverflyRule.resetState

Fixes #164

### CODE REVIEW

File: src/main/java/io/specto/hoverfly/junit/rule/HoverflyRule.java [hunk 1]
```diff
@@ -334,8 +336,32 @@ public void resetJournal() {
...
+     */
+    public void setState(final Map<String, String> state) {
+        hoverfly.setState(state);
+    }
+
+    /**
+     *  Updates state in Hoverfly.
+     */
+    public void updateState(final Map<String, String> state) {
+        hoverfly.setState(state);
```
Comment (tommysitu) [inline src/main/java/io/specto/hoverfly/junit/rule/HoverflyRule.java]: `hoverfly.updateState()` is the one.

File: src/test/java/io/specto/hoverfly/junit/api/OkHttpHoverflyClientTest.java [hunk 2]
```diff
@@ -69,6 +75,47 @@ public void shouldBeAbleToSetMode() throws Exception {
...
+
+        final StateView updateStateView = new StateView(ImmutableMap.of("key2", "value4", "key3", "value3"));
+        client.updateState(updateStateView);
+
+        StateView resultStateView = client.getState();
+
+        assertThat(resultStateView)
+                .as("get returns the state which have just been updated.")
+                .isNotNull()
+                .hasFieldOrPropertyWithValue("state", ImmutableMap.of("key1", "value1", "key2", "value4", "key3", "value3"));
```
Comment (tommysitu) [inline src/test/java/io/specto/hoverfly/junit/api/OkHttpHoverflyClientTest.java]: This is cool!

File: src/main/java/io/specto/hoverfly/junit/rule/HoverflyRule.java (hunk 1 above)
Comment (ia3andy) [reply src/main/java/io/specto/hoverfly/junit/rule/HoverflyRule.java]: Oupsy :) good catch !!

**Comment by ia3andy** [comment]
I did fix the mistake ;)

**Comment by tommysitu** [comment]
Thanks. I saw your commit, but I will wait for the other PR before merging this one.

**Pull request merged**